cep-20.md

```diff
@@ -0,0 +1,876 @@
+are installed automatically whenever the package is installed. Package
+names should follow the [package match
+specifications](https://conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications).
+
+```yaml
+requirements:
+  run:
+    - python
+    - sel(py26): argparse
+    - six >=1.8.0
+```
+
+To build a recipe against different versions of NumPy and ensure that
+each version is part of the package dependencies, list `numpy` as a
+requirement in `recipe.yaml` and use a `conda_build_config.yaml` file with
+multiple NumPy versions.
+
+### Run\_constrained
+
+Packages that are optional at runtime but must obey the supplied
+additional constraint if they are installed.
+
+Package names should follow the [package match
+specifications](https://conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications).
+
+```yaml
+requirements:
+  run_constrained:
+    - optional-subpackage =={{ version }}
+```
+
+For example, let's say we have an environment that has package "a"
+installed at version 1.0. If we install package "b" that has a
+run\_constrained entry of "a\>1.0", then mamba would need to upgrade "a"
+in the environment in order to install "b".
+
+This is especially useful in the context of virtual packages, where the
+run\_constrained dependency is not a package that mamba manages, but
+rather a [virtual
+package](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html)
+that represents a system property that mamba can't change. For example,
+a package on linux may impose a run\_constrained dependency on
+\_\_glibc\>=2.12. This is the version bound consistent with CentOS 6.
+Software built against glibc 2.12 will be compatible with CentOS 6. This
+run\_constrained dependency helps mamba tell the user that a given
+package can't be installed if their system glibc version is too old.
+
+Test section
+------------
+
+If this section exists or if there is a `run_test.[py,pl,sh,bat]` file
+in the recipe, the package is installed into a test environment after
+the build is finished and the tests are run there.
+
+### Test files
+
+Test files that are copied from the _recipe_ into the temporary test
+directory and are needed during testing.
+
+```yaml
+test:
+  files:
+    - test-data.txt
+```
+
+### Source files
+
+Test files that are copied from the _source work directory_ into the
+temporary test directory and are needed during testing (note that the
+source work directory is otherwise not available at all during testing).
+
+```yaml
+test:
+  source_files:
+    - test-data.txt
+    - some/directory
+    - some/directory/pattern*.sh
+```
+
+### Test requirements
+
+In addition to the runtime requirements, you can specify requirements
+needed during testing. The runtime requirements that you specified in
+the "run" section described above are automatically included during
+testing.
+
+```yaml
+test:
+  requires:
+    - nose
+```
```

Comment:
Just mentioning one of my personal bugbears; if we're making changes, I think specifying test dependencies under `requirements/test` makes a lot more sense than special-casing them under their own top-level heading.
e.g.
```yaml
requirements:
  build:
    - "{{ compiler('c') }}"
    - sel(linux): "{{ cdt('xorg-x11-proto-devel') }}"
  host:
    - python
  run:
    - python
    - sel(py26): argparse
    - six >=1.8.0
  run_constrained:
    - optional-subpackage =={{ version }}
  test:
    - nose
```